api/pkg/resources/openvpn/configmap.go

```
@@ -84,17 +82,22 @@ func ClientConfigConfigMap(data *resources.TemplateData, existing *corev1.Config
 	cm.Namespace = metav1.NamespaceSystem
 	cm.Labels = resources.BaseAppLabel(name, nil)
 
-	openvpnSvc, err := data.ServiceLister.Services(data.Cluster.Status.NamespaceName).Get(resources.OpenVPNServerServiceName)
+	vpnHost, vpnPort, err := data.GetMasterVpnAddress()
 	if err != nil {
 		return nil, err
 	}
+	cm.Data["config"] = getOpenVPNConfig(vpnHost, vpnPort)
+
+	return cm, nil
+}
 
-	config := fmt.Sprintf(`client
+func getOpenVPNConfig(remoteHost, remotePort string) string {
+	return fmt.Sprintf(`client
 proto tcp
 dev kube
 dev-type tun
 auth-nocache
-remote %s %d
+remote %s %s
```

Why changing to `%s` when we have a port?

---

The actual address as a host, port pair is retrieved from the `cluster-seed` configmap. There I used the more common notation of `host:port` for the "address" of the vpn endpoint. I see no added value in separating the two nor do I see value in handing the port as an integer. So in doubt I prefer less code, less fields, less conversion.

Objections?

(Previously in the cluster-controller this was indeed an integer because it was fetched as `openvpnSvc.Spec.Ports[0].NodePort`)

---

Yes, see https://github.com/kubermatic/kubermatic/pull/1956/files#r216936449

If we cast to int, we fail early in case we have invalid data instead of silently ignoring it

---

I can agree with the validation step, after passing the `host:port` notation. Sounds good.